mayastor-test/e2e/nightly/nvmf_vol/nvmf_vol_test.go

```diff
@@ -0,0 +1,124 @@
+package nvmf_vol_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	util "e2e-basic/nightly/common"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+
+	"k8s.io/client-go/deprecated/scheme"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var cfg *rest.Config
+var k8sClient client.Client
+var kubeInt kubernetes.Interface
+var k8sManager ctrl.Manager
+var testEnv *envtest.Environment
+var dynamicClient dynamic.Interface
+var defTimeoutSecs = "90s"
+
+
+func nvmfTest() {
+	fmt.Printf("running fio\n")
+	util.RunFio()
+}
+
+func TestNvmfVol(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Node Loss Test Suite")
+}
+
+var _ = Describe("Mayastor nvmf IO test", func() {
+	It("should verify an nvmf volume can process IO", func() {
+		nvmfTest()
+	})
+})
+
+var _ = BeforeSuite(func(done Done) {
+	logf.SetLogger(zap.LoggerTo(GinkgoWriter, true))
+
+	By("bootstrapping test environment")
+	useCluster := true
+	testEnv = &envtest.Environment{
+		UseExistingCluster:       &useCluster,
+		AttachControlPlaneOutput: true,
+	}
+
+	var err error
+	cfg, err = testEnv.Start()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(cfg).ToNot(BeNil())
+
+	k8sManager, err = ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: scheme.Scheme,
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	go func() {
+		err = k8sManager.Start(ctrl.SetupSignalHandler())
+		Expect(err).ToNot(HaveOccurred())
+	}()
+
+	mgrSyncCtx, mgrSyncCtxCancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer mgrSyncCtxCancel()
+	if synced := k8sManager.GetCache().WaitForCacheSync(mgrSyncCtx.Done()); !synced {
+		fmt.Println("Failed to sync")
+	}
```

Comment:
I wonder if the common bootstrapping stuff should also be moved to the common library? We are going to be doing this for every test.

Reply:
Yes, I've been thinking about doing that. It's the obvious next step.

Reply:
Now refactored